client/components/pagination/pagination-page.jsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+
+class PaginationPage extends Component {
+	static propTypes = {
+		pageNumber: React.PropTypes.node.isRequired,
+		currentPage: React.PropTypes.number.isRequired,
+		totalPages: React.PropTypes.number.isRequired,
+		pageClick: React.PropTypes.func.isRequired,
+	}
+
+	clickHandler = ( event ) => {
+		event.stopPropagation();
+		const {
+			currentPage,
+			pageClick,
+			pageNumber,
+			totalPages
+		} = this.props;
+
+		switch ( pageNumber ) {
+			case 'previous':
+				if ( currentPage - 1 < 1 ) {
+					return;
+				}
+				pageClick( currentPage - 1 );
+				break;
+			case 'next':
+				if ( currentPage + 1 > totalPages ) {
+					return;
+				}
+				pageClick( currentPage + 1 );
+				break;
+			default:
+				if ( currentPage === pageNumber ) {
+					return;
+				}
+				pageClick( pageNumber );
+				break;
+		}
+	}
+
+	render() {
+		const {
+			currentPage,
+			numberFormat,
+			pageNumber,
+			totalPages,
+		} = this.props;
+		let listClass;
```

Comment:
I'm not personally opposed to `let`, but in this case it's not really needed.
We could simply declare `const listClass` in all the switch cases where it's needed and call it a day.